### DISCUSSION TITLE
Allow lists in class

### DISCUSSION THREAD

**Discussion by TiagoCavalcante** [solidjs/solid#962] [Ideas]
In React the following pattern is pretty common:
```html
<div className={`a b c ${d}`} />
```

And then I discovered SolidJS had `classList`, but to say the truth I got a bit disappointed when I discovered it wasn't for what I expected, and it give me an idea: why not allow a list of strings along with a single string in class?
```html
<div class={["a b c", d]} />
```

**Comment by ryansolid** [comment]
This might be of interest.

https://github.com/ryansolid/dom-expressions/pull/88

This PR only looks at the runtime portion and there are compiler implications. Especially when it comes to optimization. I think to support this we'd need to move towards having only a single `class` setting prop but I need to evaluate if the de-optimizations there to the common case are too severe and if there are ways to work around it.
